### PR TITLE
[tests-only][full-ci] removing usage of a Stepdefinition inside another Stepdefinition in TrashbinContext

### PR DESCRIPTION
**Pull request by ghost**
## Description

removing the use of a stepdefinition inside another stepdefinition in `TashbinContext.php`

## Related Issue

- Related issue: https://github.com/owncloud/ocis/issues/8199

## How Has This Been Tested?

- test environment:
- test case 1:locally
- test case 2:ci

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/bootstrap/TrashbinContext.php [hunk 1]
```diff
@@ -589,16 +595,29 @@ private function convertTrashbinHref(string $href):string {
...
 	 * @param string|null $asUser
 	 * @param string|null $password
 	 *
-	 * @return int the number of items that were matched and requested for delete
-	 * @throws JsonException
-	 * @throws Exception
+	 * @return array
 	 */
-	public function tryToDeleteFileFromTrashbin(?string $user, ?string $originalPath, ?string $asUser = null, ?string $password = null):int {
+	public function atemptToDeleteFilesFromTrashBin(?string $user, ?string $originalPath, ?string $asUser = null, ?string $password = null):array {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/TrashbinContext.php:620]:
```suggestion
	public function tryToDeleteFileFromTrashbin(?string $user, ?string $originalPath, ?string $asUser = null, ?string $password = null):array {
```

File: tests/acceptance/features/bootstrap/TrashbinContext.php [hunk 2]
```diff
@@ -589,16 +595,29 @@ private function convertTrashbinHref(string $href):string {
...
 		$user = $this->featureContext->getActualUsername($user);
 		$asUser = $asUser ?? $user;
 		$listing = $this->listTrashbinFolder($user);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/TrashbinContext.php:623]: `$numItemsDeleted` can be removed
```diff
                 $listing = $this->listTrashbinFolder($user);
                $originalPath = \trim($originalPath, '/');
-		$numItemsDeleted = 0;
```

File: tests/acceptance/features/bootstrap/TrashbinContext.php [hunk 3]
```diff
@@ -642,13 +660,14 @@ public function tryToDeleteFileFromTrashbin(?string $user, ?string $originalPath
 	 * @throws Exception
 	 */
 	public function deleteFileFromTrashbin(string $user, string $originalPath):void {
-		$numItemsDeleted = $this->tryToDeleteFileFromTrashbin($user, $originalPath);
+		$responseArray = $this->atemptToDeleteFilesFromTrashBin($user, $originalPath);
 
 		Assert::assertEquals(
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/TrashbinContext.php:665]:
```suggestion
                $numItemsDeleted = \count($responseArray);
		Assert::assertEquals(
```

File: tests/acceptance/features/bootstrap/TrashbinContext.php [hunk 4]
```diff
@@ -642,13 +660,14 @@ public function tryToDeleteFileFromTrashbin(?string $user, ?string $originalPath
...
 			1,
-			$numItemsDeleted,
-			"Expected to delete exactly one item from the trashbin but $numItemsDeleted were deleted"
+			\count($responseArray),
+			"Expected to delete exactly one item from the trashbin but " . \count($responseArray) . " were deleted"
 		);
+		$this->featureContext->setResponse($responseArray[0]);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/TrashbinContext.php:670]:
```suggestion
		foreach ($responseArray as $response) {
			$this->featureContext->setResponse($response);
		}
```

**Review by saw-jan — APPROVED** [review]

File: tests/acceptance/features/bootstrap/TrashbinContext.php [hunk 5]
```diff
@@ -624,12 +642,11 @@ public function tryToDeleteFileFromTrashbin(?string $user, ?string $originalPath
 					[],
 					$user
 				);
-				$this->featureContext->setResponse($response);
-				$numItemsDeleted++;
+				$responseArray[] = $response;
```
Comment (S-Panta) [inline tests/acceptance/features/bootstrap/TrashbinContext.php:645]: Any reason for doing this?

File: tests/acceptance/features/bootstrap/TrashbinContext.php [hunk 6]
```diff
@@ -666,8 +686,15 @@ public function deleteFollowingFilesFromTrashbin(string $user, TableNode $table)
...
-			$this->deleteFileFromTrashbin($user, $path["path"]);
+			$response = $this->tryToDeleteFileFromTrashbin($user, $path["path"]);
+			$numItemsDeleted = \count($response);
 
+			Assert::assertEquals(
+				1,
+				$numItemsDeleted,
+				"Expected to delete exactly one item from the trashbin but $numItemsDeleted were deleted"
+			);
+			$this->featureContext->setResponse($response[0]);
```
Comment (S-Panta) [inline tests/acceptance/features/bootstrap/TrashbinContext.php:697]: Any reason for using index? In other place you have used for loop.

Comment (saw-jan) [reply tests/acceptance/features/bootstrap/TrashbinContext.php:697]: Ah, needs to be similar to above one
```php
foreach ($responseArray as $response) {
	$this->featureContext->setResponse($response);
}
```

Please, address the reviews in all places where there are similar codes

Comment (saw-jan) [inline tests/acceptance/features/bootstrap/TrashbinContext.php:697]:
```suggestion
			foreach ($responseArray as $response) {
			  $this->featureContext->setResponse($response);
		       }
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

**Review by saw-jan — APPROVED** [review]

**Pull request merged**